spec/cc/analyzer/issue_spec.rb

```diff
@@ -16,72 +16,44 @@ module CC::Analyzer
         },
         "remediation_points" => 10,
         "type" => "issue",
-      }.to_json
-      issue = Issue.new(output)
+      }
+    end
+
+    it "allows access to keys as methods" do
+      issue = Issue.new(output.to_json)
 
       expect(issue.respond_to?("check_name")).to eq true
       expect(issue.check_name).to eq("Rubocop/Style/Documentation")
     end
 
     describe "#fingerprint" do
       it "adds a fingerprint when it is missing" do
-        output = {
-          "categories" => ["Style"],
-          "check_name" => "Rubocop/Style/Documentation",
-          "description" => "Missing top-level class documentation comment.",
-          "location"=> {
-            "lines" => {
-              "begin" => 32,
-              "end" => 40,
-            },
-            "path" => "lib/cc/analyzer/config.rb",
-          },
-          "remediation_points" => 10,
-          "type" => "issue",
-        }.to_json
-        issue = Issue.new(output)
+        issue = Issue.new(output.to_json)
 
         expect(issue.fingerprint).to eq "9d20301efe0bbb8f87fb4eb15a71fc81"
       end
 
       it "doesn't overwrite fingerprints within output" do
-        output = {
-          "categories" => ["Style"],
-          "check_name" => "Rubocop/Style/Documentation",
-          "description" => "Missing top-level class documentation comment.",
-          "fingerprint" => "foo",
-          "location"=> {
-            "lines" => {
-              "begin" => 32,
-              "end" => 40,
-            },
-            "path" => "lib/cc/analyzer/config.rb",
-          },
-          "remediation_points" => 10,
-          "type" => "issue",
-        }.to_json
-        issue = Issue.new(output)
+        output["fingerprint"] = "foo"
+
+        issue = Issue.new(output.to_json)
 
         expect(issue.fingerprint).to eq "foo"
       end
+
+      it "uses the source fingerprint if env variable is present" do
+        ENV["CODECLIMATE_SOURCE_FINGERPRINT"] = "true"
+
+        issue = Issue.new(output.to_json)
+
+        expect(issue.fingerprint).to eq "ad79867c19494d6a1a567f804fc575ca"
+
+        ENV.delete("CODECLIMATE_SOURCE_FINGERPRINT")
```

Comment:
`ENV.delete` isn't an exception raising method, so I think it's fine as it. Doesn't seem worth it to extract as a helper method just yet.

```ruby
>> ENV.delete("CODECLIMATE_SOURCE_FINGERPRINT")
=> nil
```